File: BinarySearch.c
#include<stdio.h>
#include "cs50.h"
#include <string.h>
#include <stdlib.h>

void Binary(int min, int max, int x);

int main()
{
    int x = get_int("enter a number: ");
    int min = get_int("enter a min number: ");
    int max = get_int("enter a max number: ");

    Binary(min, max, x);
}

void Binary(int min, int max, int x)
{
    int mid = (min + max) / 2;

    if(x > min && x < max)
    {

        if(x == mid)
        {
            printf("found");
        }

        else if(x < mid)
        {
            Binary(min, mid, x);
        }

        else if(x > mid)
        {
            Binary(mid, max, x);
        }
    }
    
    else
    {
        printf("Not Found");
    }
    
}

File: BinarySearch.cs
using System;

namespace Myapp
{
    class Programme
    {
        static void Main(string[] args)
        {
            Console.Write("enter the number: ");
            int number = Convert.ToInt16(Console.ReadLine());

            Console.Write("enter the min number: ");
            int min = Convert.ToInt16(Console.ReadLine());

            Console.Write("enter the max numebr: ");
            int max = Convert.ToInt16(Console.ReadLine());

            Binary(number, min, max);
        }

        static void Binary(int number, int min, int max)
        {
            if (min != max && min < max && number > min && number < max)
            {
                int mean = (min + max) / 2;

                if (mean == number)
                {
                    Console.WriteLine("Found");
                }

                else if (mean < number)
                {
                    Binary(number, mean, max);
                }

                else
                {
                    Binary(number, min, mean);
                }
            }

            else
            {
                Console.WriteLine("There is something wrong");
            }
        }
    }
}